### PR TITLE
Azure Transition Handler

### PR DESCRIPTION
**Pull request by gkillough**

### CODE REVIEW

File: azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/service/comment/AzureWorkItemCommentService.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpMethods;
+import com.google.api.client.http.HttpRequest;
+import com.synopsys.integration.azure.boards.common.http.AzureHttpService;
+import com.synopsys.integration.azure.boards.common.http.HttpServiceException;
+import com.synopsys.integration.azure.boards.common.service.workitem.response.WorkItemCommentResponseModel;
+import com.synopsys.integration.azure.boards.common.service.workitem.response.WorkItemMultiCommentResponseModel;
+import com.synopsys.integration.azure.boards.common.util.AzureSpecTemplate;
+
+public class AzureWorkItemCommentService {
```
Comment (gkillough) [inline azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/service/comment/AzureWorkItemCommentService.java:41]: Copy/Pasted from `AzureWorkItemService`

File: azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/service/workitem/AzureWorkItemService.java [hunk 2]
```diff
@@ -118,50 +121,6 @@ public WorkItemResponseModel updateWorkItem(String organizationName, String proj
...
-        requestSpec = String.format("%s?%s=%s", requestSpec, AzureHttpService.AZURE_API_VERSION_QUERY_PARAM_NAME, "5.1-preview.3");
-        GenericUrl requestUrl = azureHttpService.constructRequestUrl(requestSpec);
-        try {
-            HttpRequest httpRequest = azureHttpService.buildRequestWithDefaultHeaders(HttpMethods.POST, requestUrl, commentTexts);
-            return azureHttpService.executeRequestAndParseResponse(httpRequest, WorkItemCommentResponseModel.class);
-        } catch (IOException e) {
-            throw HttpServiceException.internalServerError(e);
-        }
-    }
-
```
Comment (gkillough) [inline azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/service/workitem/AzureWorkItemService.java:164]: Copied to `AzureWorkItemCommentService`

File: src/main/java/com/synopsys/integration/alert/channel/azure/boards/service/AzureTransitionHandler.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        return workItemTypeState.getName();
+    }
+
+    @Override
+    public List<WorkItemTypeStateResponseModel> retrieveIssueTransitions(String workItemIdString) throws IntegrationException {
+        String organizationName = azureBoardsProperties.getOrganizationName();
+        Integer workItemId = Integer.parseInt(workItemIdString);
+        WorkItemResponseModel workItem = azureWorkItemService.getWorkItem(organizationName, workItemId);
+
+        JsonObject workItemFields = workItem.getFields();
```
Comment (bamandel) [inline src/main/java/com/synopsys/integration/alert/channel/azure/boards/service/AzureTransitionHandler.java:69]: When we have chains of calls like this, we should try to have a method in the original object that will directly give us the final result for convenience. I don't see workItem used anywhere else.

Comment (gkillough) [reply src/main/java/com/synopsys/integration/alert/channel/azure/boards/service/AzureTransitionHandler.java:69]: Yeah, I was definitely thinking about it. You pointing this out has convinced me that it needs to be done.

**Pull request merged**